operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java

```diff
@@ -119,86 +120,51 @@ protected KafkaConnect getResource() {
     }
 
     @Test
-    public void testUpdateStatus(VertxTestContext context) throws InterruptedException, ExecutionException, TimeoutException {
-        log.info("Getting Kubernetes version");
-        CountDownLatch versionAsync = new CountDownLatch(1);
-        AtomicReference<PlatformFeaturesAvailability> pfa = new AtomicReference<>();
-        PlatformFeaturesAvailability.create(vertx, client).setHandler(pfaRes -> {
-            if (pfaRes.succeeded())    {
-                pfa.set(pfaRes.result());
-                versionAsync.countDown();
-            } else {
-                context.failNow(pfaRes.cause());
-            }
-        });
-        if (!versionAsync.await(60, TimeUnit.SECONDS)) {
-            context.failNow(new Throwable("Test timeout"));
```

Comment:
I wonder if these timeouts had any special meanings and how / when the test timeouts without them. Have you done any tests around this?

Reply:
I am glad you brought this up, I should have documented this better as I raised the PR.

I have switched to using the `Checkpoint` object, this comes with the default vertx test timeout (though can be overridden as you have for several tests in other classes using e.g. (    `@Timeout(value = 2, timeUnit = TimeUnit.MINUTES)`)
I can see for some tests we might need to have to override the default timeout like above, but for the tests i refactored this was not needed.

For each of the steps i wrapped them in a `context.succeeding`, `failing`, or `verify` all of these will immediately call `context.failNow` with all relevant information if their assertions aren't met.

By using a single checkpoint at the end of a test we can gate its completion on reaching the end without throwing an exception (unless the exception is expected ofcourse).

In the event that none of the assertions fail, the checkpoint will still timeout and cause a failNow reporting that it had timed out. In cases like these the log lines are most useful for inferring where we didn't reach up to, but we can easily mitigate this, if it is not good enough, by adding in more checkpoints that are incremental throughout the test.

Reply:
Great, thanks for the explanation. 👍

Reply:
Anytime!